### PR TITLE
feat: Add support for logz.io exporter

### PR DESCRIPTION
**Pull request by yotamloe**
### Proposed Change
* Add `logzioexporter` factory  to `factories/exporters.go`
* Add reference to  `logzioexporter` docs
* Run `go mod tidy`

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Review by cpheps — APPROVED** [review]
Thanks for the contribution!

File: go.mod [hunk 1]
```diff
@@ -110,8 +111,8 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver v0.54.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zookeeperreceiver v0.54.0
 	github.com/spf13/pflag v1.0.5
-	github.com/stretchr/testify v1.7.5
-	go.opentelemetry.io/collector v0.54.0
+	github.com/stretchr/testify v1.8.0
+	go.opentelemetry.io/collector v0.55.0
```
Comment (BinaryFissionGames) [inline go.mod:115]: Our distro is currently based on v0.54.0 of the collector, could we change this PR to include version v0.54.0 of the logzio exporter instead? We'll update to v0.55.0 probably sometime later this week, or early next week.

**Review by BinaryFissionGames — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (cpheps) [reply go.mod:115]: @BinaryFissionGames @yotamloe 

I'm ok with merging this at v0.55.0. I'm working on the `v0.55.0` update today. It seems like all CI is currently passing with the mixed version.

Comment (yotamloe) [reply go.mod:115]: Sure makes sense, ill change to v0.54.0

**Review by BinaryFissionGames — APPROVED** [review]
Sounds like we're fine as is including v0.55.0, thanks @yotamloe!

**Comment by yotamloe** [comment]
Thanks, @BinaryFissionGames @cpheps 
Blazing fast response!

**Pull request merged**